go-controller/pkg/ovn/admin_network_policy_test.go

```diff
@@ -1025,6 +1077,473 @@ var _ = ginkgo.Describe("OVN ANP Operations", func() {
+				expectedDatabaseState = append(expectedDatabaseState, peerASEgressRule1v6)
+				gomega.Eventually(fakeOVN.nbClient).Should(libovsdbtest.HaveDataIgnoringUUIDs(expectedDatabaseState))
+
+				ginkgo.By("2. Create subject pod and ensure LSP and namedPorts ACLs are updated for the matching ingress rule")
+				_, err = fakeOVN.fakeClient.KubeClient.CoreV1().Pods(anpSubjectNamespaceName).Create(context.TODO(), &anpSubjectPod, metav1.CreateOptions{})
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				// The pod takes some time to get created so the first add pod event for ANP does nothing as it waits for LSP to be created - this triggers a retry.
+				// Next when the pod is scheduled and IP is allocated and LSP is created we get an update event.
+				// When ANP controller receives the pod update event, it will not trigger an update in unit testing since ResourceVersion and pod.status.Running
+				// is not updated by the test, hence we need to trigger a manual update in the unit test framework that is accompanied by label update.
+				// NOTE: This is not an issue in real deployments with actual kapi server; its just a potential flake in the UT framework that we need to protect against.
+				time.Sleep(time.Second)
+				anpSubjectPod.ResourceVersion = "5"
+				// - this is to trigger an actual update technically speaking the pod already matches the namespace of the subject
+				anpSubjectPod.Labels["rv"] = "resourceVersionUTHack"
+				_, err = fakeOVN.fakeClient.KubeClient.CoreV1().Pods(anpSubjectPod.Namespace).Update(context.TODO(), &anpSubjectPod, metav1.UpdateOptions{})
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				namedPorts["dns"] = &[]libovsdbutil.NamedNetworkPolicyPort{{L4Protocol: "udp", L3PodIP: anpPodV4IP, L3PodIPFamily: "ip4", L4PodPort: "5353"},
+					{L4Protocol: "udp", L3PodIP: anpPodV6IP, L3PodIPFamily: "ip6", L4PodPort: "5353"}}
+
+				subjectNSASIPv4, subjectNSASIPv6 = buildNamespaceAddressSets(anpSubjectNamespaceName, []string{anpPodV4IP, anpPodV6IP})
+				namedPortACLCount = 0
+				acls = getACLsForANPRules(anp)
+				pg = getDefaultPGForANPSubject(anp.Name, []string{t2.portUUID}, acls, false)
+				expectedDatabaseState = []libovsdbtest.TestData{pg, subjectNSASIPv4, subjectNSASIPv6, peerNSASIPv4, peerNSASIPv6, peerNS2ASIPv4, peerNS2ASIPv6}
+				expectedDatabaseState = append(expectedDatabaseState, getExpectedDataPodsAndSwitches([]testPod{t1, t2}, []string{node1Name})...)
+				expectedDatabaseState = append(expectedDatabaseState, peerASIngressRule0v4)
+				expectedDatabaseState = append(expectedDatabaseState, peerASIngressRule0v6)
+				expectedDatabaseState = append(expectedDatabaseState, peerASEgressRule0v4)
+				expectedDatabaseState = append(expectedDatabaseState, peerASEgressRule0v6)
+				expectedDatabaseState = append(expectedDatabaseState, peerASEgressRule1v4)
+				expectedDatabaseState = append(expectedDatabaseState, peerASEgressRule1v6)
```

Comment:
if there is a set of db objects, that are expected in every db check, can we make a separate list for them in the beginning, and then append it every time? It may simplify finding the difference between db matches

Reply:
I can try.. lmc if I can hold the AddressSets separately and mutate on the fly as needed

Reply:
another idea would be to update addrset references, so instead of 
```
subjectNSASIPv4, subjectNSASIPv6 = buildNamespaceAddressSets(anpSubjectNamespaceName, []string{anpPodV4IP, anpPodV6IP})
baseExpectedDatabaseState[0] = subjectNSASIPv4
baseExpectedDatabaseState[1] = subjectNSASIPv6
```
we could do 
```
*subjectNSASIPv4, *subjectNSASIPv6 = buildNamespaceAddressSetsObjs(anpSubjectNamespaceName, []string{anpPodV4IP, anpPodV6IP})
```
if we add a function `buildNamespaceAddressSetsObjs` that returns objects and not references, or without new functions
```
as1, as2 := buildNamespaceAddressSets(anpSubjectNamespaceName, []string{anpPodV4IP, anpPodV6IP})
*subjectNSASIPv4, *subjectNSASIPv6 = *as1, *as2
```